reactive/containerd.py

```diff
@@ -362,48 +381,38 @@ def configure_nvidia():
     status.maintenance('Installing Nvidia drivers.')
 
     dist = host.lsb_release()
-    release = '{}{}'.format(
-        dist['DISTRIB_ID'].lower(),
-        dist['DISTRIB_RELEASE']
-    )
+    os_release_id = dist['DISTRIB_ID'].lower()
+    os_release_version_id = dist['DISTRIB_RELEASE']
+    os_release_version_id_no_dot = os_release_version_id.replace('.', '')
     proxies = {
         "http": config('http_proxy'),
         "https": config('https_proxy')
     }
-    ncr_gpg_key = requests.get(
-        'https://nvidia.github.io/nvidia-container-runtime/gpgkey', proxies=proxies).text
-    import_key(ncr_gpg_key)
-    with open(
-        '/etc/apt/sources.list.d/nvidia-container-runtime.list', 'w'
-    ) as f:
-        f.write(
-            'deb '
-            'https://nvidia.github.io/libnvidia-container/{}/$(ARCH) /\n'
-            .format(release)
-        )
-        f.write(
-            'deb '
-            'https://nvidia.github.io/nvidia-container-runtime/{}/$(ARCH) /\n'
-            .format(release)
+    key_urls = config('nvidia_apt_key_urls').split()
+    for key_url in key_urls:
+        formatted_key_url = key_url.format(
+            id=os_release_id,
+            version_id=os_release_version_id,
+            version_id_no_dot=os_release_version_id_no_dot
         )
-
-    cuda_gpg_key = requests.get(
-        'https://developer.download.nvidia.com/'
-        'compute/cuda/repos/{}/x86_64/7fa2af80.pub'
-        .format(release.replace('.', '')), proxies=proxies
-    ).text
-    import_key(cuda_gpg_key)
-    with open('/etc/apt/sources.list.d/cuda.list', 'w') as f:
-        f.write(
-            'deb '
-            'http://developer.download.nvidia.com/'
-            'compute/cuda/repos/{}/x86_64 /\n'
-            .format(release.replace('.', ''))
+        gpg_key = requests.get(formatted_key_url, proxies=proxies).text
+        import_key(gpg_key)
+
+    sources = config('nvidia_apt_sources').splitlines()
```

Comment:
I'd suggest using `.split()` here like you do for the key URLs to be more forgiving.  It will split on any whitespace, including newlines, so will also support space-separated, which is easier to do from the CLI.

Reply:
You're gonna have to be more specific about what you want here. The default value for nvidia_apt_sources is:

```
deb https://nvidia.github.io/libnvidia-container/{id}{version_id}/$(ARCH) /
deb https://nvidia.github.io/nvidia-container-runtime/{id}{version_id}/$(ARCH) /
deb http://developer.download.nvidia.com/compute/cuda/repos/{id}{version_id_no_dot}/x86_64 /
```

If I `.split()` that, I get:

```
['deb', 'https://nvidia.github.io/libnvidia-container/{id}{version_id}/$(ARCH)', '/', 'deb', 'https://nvidia.github.io/nvidia-container-runtime/{id}{version_id}/$(ARCH)', '/', 'deb', 'http://developer.download.nvidia.com/compute/cuda/repos/{id}{version_id_no_dot}/x86_64', '/']
```

What do you want me to do with that?

Reply:
Oh shoot, I missed that.  Nevermind, then.